bin/publib-nuget

```diff
@@ -10,14 +10,54 @@ set -eu # we don't want "pipefail" to implement idempotency
 #
 # DIR is a directory with *.nupkg files (default is 'dist/dotnet')
 #
-# NUGET_API_KEY (required): API key for nuget
+# NUGET_API_KEY (optional): API key for nuget (ignored when using Trusted Publishers)
+# NUGET_TRUSTED_PUBLISHER (optional): set to any value to use Trusted Publisher authentication
+# NUGET_USERNAME (required when using Trusted Publishers): NuGet.org username
+# NUGET_AUDIENCE (optional): OIDC audience for token generation (defaults to https://www.nuget.org)
+# NUGET_TOKEN_SERVICE_URL (optional): NuGet token service endpoint (defaults to https://www.nuget.org/api/v2/token)
 #
 ###
 
 cd "${1:-"dist/dotnet"}"
 
-if [ -z "${NUGET_API_KEY:-}" ]; then
-    echo "NUGET_API_KEY is required"
+# Check for Trusted Publisher authentication
+if [ -n "${NUGET_TRUSTED_PUBLISHER:-}" ]; then
+    echo "Using NuGet Trusted Publisher authentication"
+    
+    if [ -z "${NUGET_USERNAME:-}" ]; then
+        echo "NUGET_USERNAME is required when using Trusted Publishers"
+        exit 1
+    fi
+    
+    # Use provided API key if available, otherwise generate one
+    if [ -n "${NUGET_API_KEY:-}" ]; then
+        echo "Using provided NUGET_API_KEY"
```

Comment:
Does a provided API key in combination with TRUSTED_PUBLISHER even make sense? Looks like we're just using the given API key? 

Isn't this equivalent to the following `if/else` chain:

```ts
if (NUGET_API_KEY) {
  publishWith(NUGET_API_KEY);
} else if (NUGET_TRUSTED_PUBLISHER) {
  publishWith(generateKey());
} else {
  error("No key given and no trusted publisher selected");
}
```

And so `NUGET_API_KEY` takes precedence? (Also means the docs are wrong then?)

Reply:
This is an attempt to support NuGet's official action to mint the required short lived token: https://github.com/NuGet/login and https://learn.microsoft.com/en-us/nuget/nuget-org/trusted-publishing

Compared to using the custom publib implementation. I'll update the docs so the intention is clear.